visionmetrics/detection/mean_ap.py

```diff
@@ -37,11 +37,15 @@ def __init__(self, box_format='xyxy', **kwargs):
 
     def update(self, predictions: List[List[List[float]]], targets: List[List[List[float]]]) -> None:
         predictions, targets = self._preprocess(predictions, targets)
+
         super().update(predictions, targets)
 
     def compute(self):
         result = super().compute()
-        return {k: v for k, v in result.items() if k in ['map', 'map_50', 'map_75', 'map_per_class', 'classes']}
+        if self.class_metrics:
+            result['map_per_class'] = {k.tolist(): v for k, v in zip(result['classes'], result['map_per_class'])}
```

Comment:
Any reason still keeping this behavior: 
- the base class returns list, while this class returns dictionary for result['map_per_class']?

Reply:
I remember in our last discussion, we agreed the need of dictionary output is mostly one kind of client need.

It does have the benefit wrt clarity, but there are ways doing it without making it a dictionary making it behave differently than the base class.

If no code change is needed, you can still check in the tests which could be helpful